### DISCUSSION TITLE
[help] How can I easlily create subtargets using `map` or `tar_map`?

### DISCUSSION THREAD

**Discussion by filipemsc** [ropensci/targets#1359] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Suppose I have the following `_targets.R` file: 

```
library(targets)
library(purrr)

tar_option_set()

subtars <- list(
  subtar1 = c("a", "b"),
  subtar2  = c("c", "d")
)

ff <- function(x,y){ paste(x,y)}

list(
  tar_target(
    name = main_tar,
    command = map(subtars, ~ ff(.x[[1]], .x[[2]])),
    cue = tar_cue(mode="always")
  )
)
```
This would be translated in this `tar_visnetwork`:

![image](https://github.com/user-attachments/assets/57f02c1c-22c0-4aa1-9935-6cc859ca6b3e)

However, I would like to have something like this:

![image](https://github.com/user-attachments/assets/9194f07d-b8c0-44b9-9beb-e6b640f48ae3)

How can I achieve this without creating the subtars individually?

(This is because in my real world application I have almost 20 subtargets and it is very important to check how each one is running individually)

**Comment by wlandau** [comment, marked as answer]
To create that branching structure in the graph, you can use static branching. Please have a look at https://books.ropensci.org/targets/static.html. Example:

```r
library(targets)
library(tarchetypes)

list(
  tar_target(
    name = main_target,
    command = "main_target"
  ),
  tar_map(
    tar_target(
      name = subtarget,
      command = paste(main_target, static_branch)
    ),
    values = list(static_branch = c("static_branch_1", "static_branch_2")),
    descriptions = NULL
  )
)
```

There is also dynamic branching, documented at https://books.ropensci.org/targets/dynamic.html

**Reply by filipemsc** [reply]
Thanks!
